File: twopointers/middleSLL.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def middleNode(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        i=head
        j=head
        while j and j.next:
            i=i.next
            j=j.next.next
            
        return i
        

File: twopointers/middleSLL.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        
        struct ListNode *i =head;
        struct ListNode *j =head;
        while (j!=NULL && j->next !=NULL){
            i=i->next;
            j=j->next->next;
            
        }
        return i;
    }
};